### DISCUSSION TITLE
New to shaders in Stride

### DISCUSSION THREAD

**Discussion by ghost** [stride3d/stride#1396] [Q&A] [answered]
i want to learn how to use shaders in stride, but i'm not sure where to start, there aren't many tutorials on it and i'm not sure what the stride equivalent is for unity shaders, so i can't use unity tutorials. if someone could point me in the right direction, i'd be grateful as i'm currently really confused on how to start.

**Comment by tebjan** [comment, marked as answer]
Hey, here are a few resources to check:
* SDSL in the manual: [Effects and shaders](https://doc.stride3d.net/4.0/en/manual/graphics/effects-and-shaders/index.html)
    * specifically: [Custom shaders](https://doc.stride3d.net/4.0/en/manual/graphics/effects-and-shaders/custom-shaders.html)
* Then create the samples: **Custom material shader** and **Custom effect**  from the launcher for simple examples
* Also create the Sample game **Space Escape** to see how to completely customize a material
* Check the project: [Stride.CustomRootRenderFeature](https://github.com/tebjan/Stride.CustomRootRenderFeature) for custom rendering

Definitely get the [Stride shader explorer](https://github.com/tebjan/Stride.ShaderExplorer) to learn from the shaders already shipping with stride and understand their inheritance hierarchy.

![image](https://user-images.githubusercontent.com/1094716/182109669-473ce76e-ca51-4008-af53-fb422d1d293d.png)

Further things to check out:
* Explanation how to override the vertex shader: https://github.com/stride3d/stride/issues/329#issuecomment-812505726
* Many of the [community projects](https://github.com/stride3d/stride/wiki/Community-Projects) use custom shaders
* in vvvv you can dynamically write Stride shaders at runtime: [Libraries/3d/Shaders]( https://thegraybook.vvvv.org/reference/libraries/3d/shaders.html)
